pennylane/pytrees.py

```diff
@@ -52,7 +104,104 @@ def register_pytree(pytree_type: type, flatten_fn: FlattenFn, unflatten_fn: Unfl
     Side Effects:
         ``pytree`` type becomes registered with available backends.
 
+    .. seealso:: :func:`~.flatten`, :func:`~.unflatten`.
+
     """
 
+    _register_pytree_with_pennylane(pytree_type, flatten_fn, unflatten_fn)
+
     if has_jax:
         _register_pytree_with_jax(pytree_type, flatten_fn, unflatten_fn)
+
+
+@dataclass(repr=False, frozen=True)
+class PyTreeStructure:
+    """A pytree data structure, holding the type, metadata, and child pytree structures.
+
+    >>> op = qml.adjoint(qml.RX(0.1, 0))
+    >>> data, structure = qml.pytrees.flatten(op)
+    >>> structure
+    PyTree(AdjointOperation, (), [PyTree(RX, (<Wires = [0]>, ()), [Leaf])])
+
+    A leaf is defined as just a ``PyTreeStructure`` with ``type=None``.
+    """
+
+    type: Optional[type]
+    """The type corresponding to the node. If ``None``, then the structure is a leaf."""
+
+    metadata: Metadata
+    """Any metadata needed to reproduce the original object."""
+
+    children: list["PyTreeStructure"]
+    """The children of the pytree node.  Can be either other structures or terminal leaves."""
+
```

Comment:
It'd be good to have a property to make the leaf definition explicit:
```suggestion
    @property
    def is_leaf(self) -> bool:
        return self.type is None
```